src/hotspot/share/gc/shenandoah/shenandoahFreeSet.hpp

```diff
@@ -340,7 +340,9 @@ class ShenandoahFreeSet : public CHeapObj<mtGC> {
   void move_regions_from_collector_to_mutator(size_t cset_regions);
 
   void recycle_trash();
+  // log status, assuming lock has alredy been acquired by the caller.
```

Comment:
"already". Also see previous comment about removing enablement check in `log_status` and asserting. Will then need to check other callers and retest.

Reply:
We should keep it, finish_rebuild also call log_status since it already holds the lock, removing  enablement check will affect the log behavior in finish_rebuild.

Reply:
I got your point, but we have two enablement checks, one for Info one for Debug, we can't remove it from finish_rebuild.